### ISSUE TITLE
382 medium of performance

### ISSUE THREAD

**Issue by CECSpecialistI** [uwlib-cams/MARC2RDA#171]
https://github.com/uwlib-cams/MARC2RDA/blob/main/Working%20Documents/3XX.csv

**Comment by CECSpecialistI** [comment]
@AdamSchiff @CatalogerCate Alice and I just started working on the 382 field and have some questions. What is the difference between "medium of performance" and "medium of performance for musical content of representative expression"? Also, what is indicated by the second indicator here, "Access control"? The options are  # for no information provided, 0 for not intended for access, and 1 for intended for access. What is "intended for access" referring to here? Thank you for sharing your expertise!

**Comment by CatalogerCate** [comment]
We haven’t really started using the second one yet.  The whole representative expression is a “new rda” concept that hasn’t been implemented.  But basically, the medium of performance is what you have in hand, let’s say a recording of a sonata with flute and piano.  The representative expression would be the instrumentation of what is considered to be the “most authoritative version” of the work, which might be violin and piano.  Basically the cataloger would decide the instrumentation of the representative expression regardless of what they have in hand.  They could be the same if the medium of performance is the same at the “representative expression”.

The second indicated of the 382 just allows systems to ignore the 382 if they don’t want to do the programming to get the display to work.  In OCLC was always use 1 for intended for access (will be used for access), but if your system doesn’t want to figure out how to display this field, you might locally set it to 0.  That’s because this field takes lots of work to get it to display in a meaningful way to the public.

Here’s an example.

MARC:  382 01  baritone voice ǂn 1 ǂa piano ǂn 1 ǂs 2 ǂ2 lcmpt
Primo turns that into: For baritone voice; piano. Total performers: 2.

The primo display isn’t great, but it’s more clear than the 382 MARC field.

Does that help?

Cate

**Comment by CECSpecialistI** [comment]
Yes, that helps a lot! Thank you, Cate!

**Comment by CECSpecialistI** [comment]
Does it make a difference to the display whether it is full vs. incomplete? (1st indicator 0 vs. 1?)

**Comment by CatalogerCate** [comment]
Nope, the first indicator allows you to code just part of the instrumentation.  For instance, you might have a recordings with a soprano, but there is no indication what the instrumentation is for the accompanying ensemble, so you could put “soprano voice’ in and not try to figure out what other instruments are that are accompanying her. Or, perhaps it varies on every track and it’s not possible to added 20 382s for all the different combinations of instruments.  So, you just indicate the important soloist.

Cate

**Comment by AdamSchiff** [comment]
medium of performance for musical content of representative expression is a work attribute and would be the "canonical" medium of performance that the work was originally conceived and written for.

medium of performance is an expression attribute and could be recorded for the original expression of the work or for arrangements for different medium of performance.

Here's an example:

WORK AUTHORITY:

100 1# Andriessen, Louis, ǂd 1939-2021. ǂt Elegy, ǂm cello, piano
046 ## ǂk 1957 ǂ2 edtf
382 2# cello ǂn 1 ǂa piano ǂn 1 ǂs 2 ǂ2 lcmpt
670 ## Andriessen, L. Elegy, cello or double bass and piano, c2002: ǂb t.p. (Elegy, cello or double bass and piano) t.p. verso (composed in 1957 for violoncello and piano; arr. in the early 1990s for double bass and piano)

EXPRESSION AUTHORITY:

100 1# Andriessen, Louis, ǂd 1939-2021. ǂt Elegy, ǂm cello, piano; ǂo arranged
046 ## ǂk 199X ǂ2 edtf
382 0# double bass ǂn 1 ǂa piano ǂn 1 ǂs 2 ǂ2 lcmpt
670 ## Andriessen, L. Elegy, cello or double bass and piano, c2002: ǂb t.p. (Elegy, cello or double bass and piano) t.p. verso (composed in 1957 for violoncello and piano; arr. in the early 1990s for double bass and piano)

In a bib record for this manifestation cited in the 670 above:

100 1# Andriessen, Louis, ǂd 1939-2021.
240 10 Elegy, ǂm cello, piano
245 10 Elegy, cello or double bass and piano / ǂc Louis Andriessen ; version for double bass edited by Quirijn van Regteren Altena.
260 ## [London?] : ǂb Boosey & Hawkes, ǂc ©2002.
300 ## 1 score (5 pages) + 2 parts ; ǂc 31 cm
382 2# cello ǂn 1 ǂa piano ǂn 1 ǂs 2 ǂ2 lcmpt
382 0# ǂ3 Original ǂa cello ǂn 1 ǂa piano ǂn 1 ǂs 2 ǂ2 lcmpt
382 0# ǂ3 Arrangement ǂa double bass ǂn 1 ǂa piano ǂn 1 ǂs 2 ǂ2 lcmpt

Does this help?

Adam

Adam L. Schiff
Principal Cataloger
University of Washington Libraries
(206) 543-8409
***@***.***
________________________________

**Comment by AdamSchiff** [comment]
Well ideally if it is incomplete, the display would be labeled as such to indicate that it isn't the complete MOP.

Adam L. Schiff
Principal Cataloger
University of Washington Libraries
(206) 543-8409
***@***.***
________________________________

**Comment by CECSpecialistI** [comment]
Thank you @AdamSchiff ! Will reflect the work vs. expression properties and incomplete vs. complete in the mapping draft.

**Comment by CECSpecialistI** [comment]
Thank you for explaining the MARC indicators, @CatalogerCate ! Very handy to have a music cataloger to consult, I was lost.

**Comment by CatalogerCate** [comment]
Happy to help!

Cate

**Comment by CECSpecialistI** [comment]
@junghaelee , @cwarc and I are struggling with the conditions for this mapping. Does Primo have norms rules that control display for the MARC 382 field that our mapping could borrow? How does our system generate the human-readable displays in Primo?

**Comment by junghaelee** [comment]
We display subfields in of the 382 field as they appear. We insert punctuation as needed in case some subfields are missing. Here is the display norm rules of the 382 field, which includes detailed information of the purpose of the rule.

https://github.com/Orbis-Cascade-Alliance/ve-norm-rules/blob/main/Display_lf87_Instrumentation.txt

**Comment by CECSpecialistI** [comment]
@pan-zhuo @gerontakos would it be possible to simplify this mapping significantly, and rather than trying to express how we want string outputs formatted using conditions in the spreadsheet, to write the transform code based on [these norm rules](https://github.com/Orbis-Cascade-Alliance/ve-norm-rules/blob/main/Display_lf87_Instrumentation.txt)? If you take a look at [the spreadsheet](https://docs.google.com/spreadsheets/d/1PE1UNwxxxcNdV4wTZqUNYYvbtTrPt4XOpSS5A8UT_x4/edit?usp=sharing) I've been working on, the conditions are getting extremely granular, all for the sake of outputting a string value with the right words in the right order. All the conditions/transformation notes are in progress and most of them aren't correct.

**Comment by CatalogerCate** [comment]
I’m not sure quite how to read this norm rules, but I think the most important thing in the 382 is that order matters!  When a $a, b, and n appear in the 382 they go together.  There may well be multiple instances of a, b, p, and n, but they absolutely have to stay clumped together to make any sense.  If somehow the $a’s all got put together, then the $b’s etc., it would be useless.

For example:

$a flute $b piccolo $n 2 $a oboe $b English horn $n 1

The first $b has to go with the first a and the second b needs to stay with the second a.  The first $n must stay with the first a, and the second $n must stay with the second $a.

Cate

**Comment by CECSpecialistI** [comment]
Right, the endless combinations of orders of subfields that make a difference in meaning is what is making the mapping so complex. If we could just map to properties based on indicators and then use the Primo norm rules to output the string values it would save a lot of work.

**Comment by AdamSchiff** [comment]
I completely concur with Cate, and it is for this reason that the URIs in MARC group said that you cannot include URIs for the individual instruments in the 382 field since the order is important.

I do have a question for you though Cate:

$a flute $b piccolo $n 2 $a oboe $b English horn $n 1

There are two flutes, but are there two piccolos that double them, or just 1 piccolo doubling a flute?

Adam

Adam L. Schiff
Principal Cataloger
University of Washington Libraries
(206) 543-8409
***@***.***

________________________________

**Comment by CECSpecialistI** [comment]
Right, I don't think IRI's will be appropriate here, I think the notes we have displaying in Primo are the goal, really.

**Comment by CatalogerCate** [comment]
Sorry! Should be:

$a flute $2 $d piccolo $n 1 $a oboe $n 1 $d English horn $n 1

So, 2 flutes, one doubling piccolo and one oboe doubling English horn!  And, $d NOT $b.  Geez.

Cate

**Comment by AdamSchiff** [comment]
Got it, thanks for the correction. Much clearer with $n's after each of the other subfields.

Adam L. Schiff
Principal Cataloger
University of Washington Libraries
(206) 543-8409
***@***.***
________________________________

**Comment by CECSpecialistI** [comment]
faceting rules https://github.com/Orbis-Cascade-Alliance/ve-norm-rules/blob/main/Facet_lf59_MusicMediumPerformanceStatement.txt

**Comment by CECSpecialistI** [comment]
@AdamSchiff @GordonDunsire @cwarc @gerontakos I got a response to the email inquiry we discussed at last week's meeting:

"The documentation that Gordon refers to is a series of email conversations that we had regarding how a complete rendering of the medium of performance of musical content for an expression could be expressed in a linked data friendly way in the context of the RDA element set.

The issue is how do you associate the instrument with the quantity required and how does one take the instrument/quantity combo and associate it with the other combinations in the expression. The solution, if one could call it that, would be to have properties for the instrument required and the number of instruments required; a class whose sole purpose would be to relate the instrument to the quantity; and then another property that would order and format the instrument/quantities according to a string encoding scheme:

ex:NumberOfInstrumentsStatement [
rdae:mediumOfPerformanceOfMusicalContent "violin" .
ex:numberOfInstruments "2" .
] ,
[
rdae:mediumOfPerformanceOfMusicalContent "viola" .
ex:numberOfInstruments "1" .
] ,
[
rdae:mediumOfPerformanceOfMusicalContent "cello" .
ex:numberOfInstruments "1" .
] .

<ex:mediumOfPerformanceOfMusicalContentStatement> "violin (2) viola (1) cello (1)" .

But notice that for a relatively simple MOP like this one, we are generating 3 separate blank nodes, which is less than ideal. This also has the added problem of adding a new class to RDA, which we want to avoid at all costs.

The solution that we would probably prefer, but have not yet pursued, is that rdae:mediumOfPerformanceOfMusicalContent would probably just be fine with having the value for the instrument/voice and that a MOP statement element could provide the context and the quantities, without having to have a separate element for the number of instruments:

<Expression1>
rdae:mediumOfPerformanceOfMusicalContent "violin" , "viola" , "cello" .
ex:mediumOfPerformanceOfMusicalContentStatement "violin (2) viola (1) cello (1)" .

For ex:mediumOfPerformanceOfMusicalContentStatement, the string encoding scheme could specify to follow the instrument by a quantity, without having to have a separate element for the quantity.

With regards to your mapping from MARC field 382 to RDA in the current environment, I am mostly in agreement with your map. A single instrument or voice is acceptable in the rdae:mediumOfPerformanceOfMusicalContent element or its work counterpart. It should also be said that the element could also be used to express a full medium of performance as an unstructured note, so it is not necessary to use "has note on expression":

382 0# $a violin $n 2 $a viola $n 1 $a cello $n 1 $s 4
could be mapped as

rdae:mediumOfPerformanceOfMusicalContent "violin" , "viola" , "cello" , "violin (2), viola (1), cello (1) (total performers: 4)" .
(or alternatively)
rdae:mediumOfPerformanceOfMusicalContent "violin (2)" , "viola (1)" , "cello (1)" , "violin (2), viola (1), cello (1) (total performers: 4)" .

or if you want to use the IRIs for the MOP terms and a string for the combined statement:

rdae:mediumOfPerformanceOfMusicalContent lcmpt:violin , lcmpt:viola , lcmpt:cello , "violin (2), viola (1), cello (1) (total performers: 4)" .

More than happy to talk further about this,
Damian


Damian Iseminger
RDA Technical Team Liaison Officer"

**Comment by CECSpecialistI** [comment]
@CatalogerCate ,

We've copied over the Alliance norms rules for display of the 382 field as a note for the MARC2RDA mapping. Part of these rules indicate that certain aspects of the 382 be displayed using square brackets. Because square brackets can sometimes have specific meanings in MARC data that aren't applicable here, @GordonDunsire has suggested that we replace them with parenthesis. @pan-zhuo mentioned that MLA has guidance on punctuation for display of 382 fields, which could be a reason to leave the square brackets as they are. Do you have any thoughts on this? How about @AdamSchiff ?

**Comment by pan-zhuo** [comment]
Here's the MLA/MOUG Display Preferences document
https://musicoclcusers.org/wp-content/uploads/WCD_Medium_Report_201504291.pdf

> 9. Insert square brackets around content in $v (58%)
> &nbsp;&nbsp;&nbsp;&nbsp;382 01 mixed chorus $v SATB
> &nbsp;&nbsp;&nbsp;&nbsp;mixed chorus [SATB]
> &nbsp;&nbsp;&nbsp;&nbsp;Note: 41% preferred parentheses; square bracket would provide differentiation from numeric values

**Comment by CatalogerCate** [comment]
If you look at the most recent MLA best practices for the 382 (https://cmc.wp.musiclibraryassoc.org/wp-content/uploads/sites/5/2023/03/BestPracticesforUsingLCMPT_1-5_20230301.pdf), you will not see any indication that square brackets or parentheses should be used.  Any use of that kind of thing is purely for display and NOT for use in the MARC field.  When mapping, I would not indicate anything should be put in brackets or parentheses. Cate

**Comment by CECSpecialistI** [comment]
Thanks Cate!

**Comment by sarahruska** [comment]
Reviewed the spreadsheet and not making any changes. Since already coded, will mark as done and close.

**Comment by cspayne** [comment]
I missed this one! We need to update this mapping to match our decisions on mapping 3XX attributes. Assigning myself to do this and update the code.

**Comment by cspayne** [comment]
Hi @sarahruska are you willing/able to do a quick review of this again? I added two new rows with conditions for when $2 is present. This is based on our discussions of 3XX fields with $2 and shouldn't take too long to review. 

Essentially, we will see if the value is part of LC's performance medium vocabulary. If it is, we use that IRI. If it's not we have two options: If a $2 is present, we mint a concept for the value with a skos:inScheme of the source. If no $2 is present, we use a string value.

**Comment by cspayne** [comment]
> Essentially, we will see if the value is part of LC's performance medium vocabulary. If it is, we use that IRI. If it's not we have two options: If a $2 is present, we mint a concept for the value with a skos:inScheme of the source. If no $2 is present, we use a string value.

Actually, looking at this vocabulary, it is not one that we can do lookups to because an xml version is not available on LC (that I can find). I've updated the mappings to reflect that.

**Comment by CatalogerCate** [comment]
LC medium of performance is in the LC linked system.  Look here:



Cate Gerhart
Head, Monographic Cataloging Unit
University of Washington Libraries
Box 352900
Seattle, WA  98195

206 685-2827 (office)
206 372-5046 (cell)
________________________________

**Comment by CatalogerCate** [comment]
Sorry.... here

https://id.loc.gov/search/?q=&q=cs%3Ahttp%3A%2F%2Fid.loc.gov%2Fauthorities%2FperformanceMediums


Cate Gerhart
Head, Monographic Cataloging Unit
University of Washington Libraries
Box 352900
Seattle, WA  98195

206 685-2827 (office)
206 372-5046 (cell)
________________________________

**Comment by cspayne** [comment]
Thanks Cate! The problem in this phase is that there doesn't seem to be a downloadable XML version of the vocabulary. We ran into problems querying LC too fast so our work around is to download the files we need to lookup values in, in bulk, at the beginning of the transformation.

**Comment by CatalogerCate** [comment]
Ah, well this vocabulary is fairly small, only a few hundred terms, so maybe it would be quicker to query?

Cate


Cate Gerhart
Head, Monographic Cataloging Unit
University of Washington Libraries
Box 352900
Seattle, WA  98195

206 685-2827 (office)
206 372-5046 (cell)
________________________________

**Comment by pan-zhuo** [comment]
Download is available here under Alternate Formats

https://id.loc.gov/authorities/performanceMediums.html

**Comment by cspayne** [comment]
> Download is available here under Alternate Formats
> 
> https://id.loc.gov/authorities/performanceMediums.html
> […](#)

Looking at those download files, they only include the top-level scheme members. In order to download and access all of the terms, we would need to also download the separate files for 'performer', 'ensemble', and 'visuals', and then all the files for their narrower terms...

I don't think this is in scope for Phase 1, but we should revisit in Phase 2.

**Comment by cspayne** [comment]
For phase 1 we will mint skos:concepts for each term with the source IRI provided as a skos:inScheme value, like we do with other similar fields.

**Comment by sarahruska** [comment]
> Hi [@sarahruska](https://github.com/sarahruska) are you willing/able to do a quick review of this again? I added two new rows with conditions for when $2 is present. This is based on our discussions of 3XX fields with $2 and shouldn't take too long to review.

@cspayne - I reviewed and didn't make any changes. 

> I don't think this is in scope for Phase 1, but we should revisit in Phase 2.

I'll create a new issue to evaluate incorporating LC Alternate Formats in Phase 2.

> For phase 1 we will mint skos:concepts for each term with the source IRI provided as a skos:inScheme value, like we do with other similar fields.

I think Decisions Index [II.C.2 $0's and $1's for 3XX fields](https://github.com/uwlib-cams/MARC2RDA/wiki/Decisions-Index#iic2-0s-and-1s-for-3xx-fields) needs some clarification - If I'm following, I think we need to update "II.C.2.a We will record an external IRI in a subfield $0 or $1 from a 3XX field as the object of a triple of an attribute property. 2024-07-16" to reflect https://github.com/uwlib-cams/MARC2RDA/issues/439#issuecomment-2476991289:
 
> We use id.loc.gov IRIs as schemes for nomens and for skos:Concepts. For vocabulary terms such as in 3XX fields, we perform lookups to retrieve the IRI for that term if possible based on the source. Otherwise, we mint a skos:Concept if a source is provided that we cannot do a lookup for. If there is no source, we use a string value. We do not do string values with an rdf:datatype, which was the previous method.

Essentially, II.C.2.a is overwritten by the other direction in II.C.1 and II.D.2.b. 

If that's on-track, I'll update the Decisions Index - but please set me straight if that's incorrect!

**Comment by cspayne** [comment]
Hi Sara, thanks for taking a look!

Looking at Decision II.C.2.a, it is correct as is. If a $0 (containing http) or a $1 is present in the 3XX field, we will use it. What I'm discussing here relates to II.K, specifically II.K.2 about looking up IRIs for $2 values. These IRIs are used as sources for a minted concept.

There is one other aspect of 3XX fields that I don't think ever made it in to the decisions index, which is that when vocabulary terms or codes are from RDA vocabularies or LC's cloned RDA vocabularies, the terms can often be looked up and an IRI retrieved to be used as the object of the attribute property. This is done by initiating a bulk download of files for these vocabularies at the start of the transform and performing local lookups to those files during the transform. 
It is only when we cannot do this, that we mint a skos:concept and use the $2 source as skos:inScheme. 

382 is a case where this cannot occur, because there is not one XML file that contains a list of all terms within the vocabulary used, but for many other fields, this is present in the mapping and in the code, so we must mint skos:concepts

Let me know if I can make this any clearer or answer any questions!

**Comment by pan-zhuo** [comment]
Weird they would publish this way and not provide bulk downloads like
LCGFT/LCDGT. Sinopia used to cache LC vocabularies but they switched to
direct query now. Good thing that LC IRIs are predictable by appending the
right extension ("skos.rdf") to base IRIs, and that LCMPT is hierarchical,
so you could use a recursion program to get everything. See also
https://id.loc.gov/techcenter/downloads.html. Definitely could ask LC to
publish bulk download as well.

**Comment by AdamSchiff** [comment]
http://id.loc.gov/authorities/performanceMediums

Adam L. Schiff
Principal Cataloger
University of Washington Libraries
Box 352900
Seattle, WA 98195-2900
aschiff @ uw.edu
________________________________

**Comment by CatalogerCate** [comment]
The 382 is a few different vocabulary than the lcgft and lcdgt.  The 382 frequently had multiple terms from the vocabulary that work together to create a string that represents the instrumentation for a piece of music.  She one 382 will point to one, two, three, or more terms in the list.  It’s hard to explain, but the use of the list is just very different than the other two lists.

Cate

Cate Gerhart
Head, Monographic Cataloging Unit
University of Washington Libraries
Box 352900
Seattle, WA  98195

206 685-2827 (office)
206 372-5046 (cell)

**Comment by AdamSchiff** [comment]
I don't know why there isn't a bulk download for LCMPT.  It's worth asking LC NDMSO why it isn't available: ***@***.***

Adam

Adam L. Schiff
Principal Cataloger
University of Washington Libraries
Box 352900
Seattle, WA 98195-2900
aschiff @ uw.edu
________________________________
